keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt

```diff
@@ -26,21 +26,30 @@ import org.springframework.stereotype.Component
 class DebianArtifactSupplier(
   override val eventPublisher: EventPublisher,
   private val artifactService: ArtifactService
-) : ArtifactSupplier<DebianArtifact, DebianSemVerVersioningStrategy> {
+) : ArtifactSupplier<DebianArtifact, NetflixSemVerVersioningStrategy> {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
   override val supportedArtifact = SupportedArtifact("deb", DebianArtifact::class.java)
 
   override val supportedVersioningStrategy =
-    SupportedVersioningStrategy("deb", DebianSemVerVersioningStrategy::class.java)
+    SupportedVersioningStrategy("deb", NetflixSemVerVersioningStrategy::class.java)
+
+  override fun publishArtifact(artifact: PublishedArtifact) {
+    if (artifact.hasReleaseStatus()) {
+      super.publishArtifact(artifact)
+    } else {
+      log.debug("Ignoring artifact event without release status: $artifact")
+    }
+  }
 
   override fun getLatestArtifact(deliveryConfig: DeliveryConfig, artifact: DeliveryArtifact): PublishedArtifact? =
     runWithIoContext {
-      artifactService
-        .getVersions(artifact.name)
+      artifactService.getVersions(artifact.name, artifact.statusesForQuery, DEBIAN)
```

Comment:
This was previously looking up versions for all statuses, not only the ones configured in the delivery artifact.